File: app/src/main/java/com/canli/oya/traininventory/viewmodel/ChosenTrainFactory.java
package com.canli.oya.traininventory.viewmodel;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import com.canli.oya.traininventory.data.TrainDatabase;

public class ChosenTrainFactory extends ViewModelProvider.NewInstanceFactory {

    private final TrainDatabase mDb;
    private final int mTrainId;

    public ChosenTrainFactory(TrainDatabase database, int trainId) {
        mDb = database;
        mTrainId = trainId;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new ChosenTrainViewModel(mDb, mTrainId);
    }
}

File: app/src/main/java/com/canli/oya/traininventoryroom/viewmodel/ChosenTrainFactory.kt
package com.canli.oya.traininventoryroom.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

import com.canli.oya.traininventoryroom.data.TrainDatabase

class ChosenTrainFactory(private val mDb: TrainDatabase, private val mTrainId: Int) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return ChosenTrainViewModel(mDb, mTrainId) as T
    }
}